File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        max_so_far = 1
        consecutive_sequence_len = 1
        for num in nums_set:
            if num - 1 in nums_set:
                continue
            else:
                looking_for = num + 1
                while looking_for in nums_set:
                    consecutive_sequence_len += 1
                    looking_for += 1
                max_so_far = max(max_so_far, consecutive_sequence_len)
        return max_so_far

File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
#include <climits>

class Solution {
public:
    int maxSubArray(vector<int> &nums) {
        int max_so_far = INT_MIN;

        for (int i = 0; i < nums.size(); i++) {
            int sum_from_i = 0;
            for (int j = i; j < nums.size(); j++) {
                sum_from_i += nums[j];
                if (sum_from_i > max_so_far) {
                    max_so_far = sum_from_i;
                }
            }
        }
        return max_so_far;
    }
};
